### ISSUE TITLE
'bibio-files' variable not set when 'bibliography' field is specified in YAML metadata

### ISSUE THREAD

**Issue by andrewmwhite** [jgm/pandoc#1661]
It appears that the 'bibio-files' template variable is not set when the 'bibliography' field is specified in YAML metadata (without '--bibliography' on the command line).

mwe.md:
```
---
bibliography: mwe.bib
...

Pandoc is great [@pandoc].
```
mwe.bib:
```
@misc{pandoc,
  title = {Pandoc: A Universal Document Converter},
  howpublished = {http://johnmacfarlane.net/pandoc},
}
```

Output with '--bibliography' (as expected):
```
> pandoc --biblatex --filter pandoc-citeproc --bibliography=mwe.bib -t latex mwe.md | grep bib
\usepackage{biblatex}
\bibliography{mwe}
\printbibliography
```

Should be the same, but the '\bibliography{mwe}' line is missing:
```
> pandoc --biblatex --filter pandoc-citeproc -t latex mwe.md | grep bib
\usepackage{biblatex}
\printbibliography
```

Side note: the '\bibliography' command is deprecated (according to the biblatex manual) and has been replaced with '\addbibresource'.

Version info:
```
> pandoc --version                                                                                                                                                                                           
pandoc 1.13.1
Compiled with texmath 0.8, highlighting-kate 0.5.9.
<snip>
> pandoc-citeproc --version                                                                                                                                                                                pandoc-citeproc 0.6
```

**Comment by azrdev** [comment]
Can confirm that.
```
> pandoc --version
pandoc 1.12.3.3
...
> pandoc-citeproc --version
pandoc-citeproc 0.3.1
```

**Comment by lierdakil** [comment]
Actually, this is expected behavior and not in any way a bug, according to README and changelog. It may be a reasonable enhancement though.

**Comment by jgm** [comment]
See also #1849, which should be fixed at the same time.

**Comment by jgm** [comment]
I think what makes sense for this issue and #1849 is the following:

[ ] Replace `biblio-files` with `bibliography` in the latex and beamer templates.
[ ]  Remove `biblio-files` specific code from pandoc.hs.
[ ]  Remove refs to `biblio-files` in README.
[ ]  Change pandoc.hs so it sets implicit `-F pandoc-citeproc` only if the command line option `--bibliography` is used, and not if there is a metadata `bibliography` field.  (Currently it just checks metadata, because the command line option sets the metadata, and the way it's currently done there's no way to tell whether the option was set by command line or in actual metadata.  This can be fixed by adding a new field to Opt.

**Comment by lierdakil** [comment]
@jgm I don't think that if we remove `biblio-files`, `--bibliography` should automatically imply `-F pandoc-citeproc` anymore, since using it with `--biblatex`/`--natbib` would be equally feasible option, and using `-V bibliography`/`-M bibliography` is somewhat counterintuitive.

`pandoc-citeproc` may be a reasonable default, if nothing else is specified, but should not be forced.

**Comment by jgm** [comment]
It's already the case that `--bibliography` doesn't imply `-F pandoc-citeproc` if you specify `--biblatex` or `--natbib` on the command line as well.

**Comment by lierdakil** [comment]
Oh, ok. I missed that.

**Comment by jgm** [comment]
Closed by f0817e19527dc366339c070ac814e71f6483f6f6 and c6debff1f898b0fd6a36efe8397dbfd18b84cacf

**Comment by bscottm** [comment]
For LaTeX: bad idea to escape underscores in file names (was doing some independent testing).

\addbibresource{foo_bar.bib}  % works
\addbibresource{foo\_bar.bib} % no data source
